### PR TITLE
New GRIB-CF mappings

### PR DESCRIPTION
**Pull request by trexfeathers**
Linked: SciTools/iris#4859 - Julian Heming is aiming for some additional GRIB-to-PP mappings.

Checking Julian's suggestions against the [WMO Codes Registry](http://codes.wmo.int/grib2/codeflag/4.2) allowed me to get a `long_name` for these new mappings, which I see isn't usually done. Hopefully that's not a breaking addition.

This is fresh territory for me so I'm not 100% confident. Please can the reviewer be vigilant!

I also need Julian's eyes on this before merging. I don't think he's on GitHub but I will be contacting personally.

### CODE REVIEW

File: iris_grib/_grib_cf_map.py [hunk 1]
```diff
@@ -63,27 +63,36 @@
...
     G2Param(2, 0, 1, 0): CFName('specific_humidity', None, 'kg kg-1'),
     G2Param(2, 0, 1, 1): CFName('relative_humidity', None, '%'),
     G2Param(2, 0, 1, 2): CFName('humidity_mixing_ratio', None, 'kg kg-1'),
     G2Param(2, 0, 1, 3): CFName(None, 'precipitable_water', 'kg m-2'),
     G2Param(2, 0, 1, 7): CFName('precipitation_flux', None, 'kg m-2 s-1'),
+    G2Param(2, 0, 1, 9): CFName('stratiform_rainfall_amount', 'Large-scale precipitation (non-convective)', 'kg m-2'),
+    G2Param(2, 0, 1, 10): CFName('convective_rainfall_amount', 'Convective precipitation', 'kg m-2'),
     G2Param(2, 0, 1, 11): CFName('thickness_of_snowfall_amount', None, 'm'),
     G2Param(2, 0, 1, 13): CFName('liquid_water_content_of_surface_snow', None, 'kg m-2'),
+    G2Param(2, 0, 1, 15): CFName('stratiform_snowfall_amount', 'Large-scale snow', 'kg m-2'),
```
Comment (trexfeathers) [inline iris_grib/_grib_cf_map.py:76]: Julian: your suggestion here was a mapping of `0, 1, 13` to STASH `m01s04i202`. I have instead proposed a mapping that would link `0, 1, 15`.

STASH `m01s04i202` is already mapped to the CF standard name `stratiform_snowfall_amount`, and existing similar mappings for other `stratiform_...` standard names suggest this is the correct course.

Thoughts?

**Comment by trexfeathers** [comment]
I think the test failure may be due to upstream Iris changes. Will look tomorrow.

File: iris_grib/_grib_cf_map.py (hunk 1 above)
Comment (trexfeathers) [reply iris_grib/_grib_cf_map.py:76]: Julian has confirmed for me that `0, 1, 15` is the correct mapping.

**Comment by trexfeathers** [comment]
I expect these mappings are all fine. To avoid staleness I propose reviewing and merging at the end of September.

**Comment by trexfeathers** [comment]
Julian has approved this as working for their tests 👍

**Comment by trexfeathers** [comment]
Needs #299 to fix the tests.

**Review by bjlittle — APPROVED** [review]
@trexfeathers LGTM :+1:

**Pull request merged**